### ISSUE TITLE
v3.2.0: DB Migration from Firebase -> Supabase

### ISSUE THREAD

**Issue by xmliszt** [xmliszt/taboo-ai#263]
## Why do we migrate?

- Take advantages of Postgres SQL
- Cheaper in the long run?
- Come with RLS and Auth (We can get rid of the annoying Google popup error blocked by browser)
- Relational DB more organised
- Integration with NextJS server actions and other fancy tech

---
## Tasks
- [x] Setup supabase DB and schema
- [x] Preview: Migrate users (trigger: if user already exists when sign up, update the row)
- [x] Preview: Migrate levels
- [x] Preview: Migrate words
- [x] Preview: Migrate games & scores
- [x] Preview: Migrate subscriptions
- [x] Preview: Migrate app and levels stats from realtime db
- [x] Migrate firebase cloud functions utilities
- [x] Auth migration to supabase
- [x] Code base change to use supabase
- [x] Supabase view & rpc for fetching & inserting complex data instead of so many api calls
- [x] Realtime db fetch app stats
- [x] Stripe subscription handle user purposedly delete account to obtain free trials again
- [x] login issue for new user
- [x] Pricing page layout issue on phone
- [x] Pro plan add "ai suggestion", and all use gemini pro
- [x] nuke openai
- [x] pricing page free plan "downgrade" text not correct 
- [x] CICD - supabase deployment
- [x] ⚠️ Resolve all `TODO` items 
- [x] Test and fix all bugs from migration
- [x] Remove mention about curated topic (as we gonna use separate PR to push the new feature)
- [x] One last check make sure migration script align 100%
- [x] nuke contact-me, move substack pub to /publication route separately, move links (buymecoffee, join discord, open source) into side bar, keep home screen clean. if user not pro user, show pricing plan below
- [ ] Production: Migrate users
- [ ] Production: Migrate levels
- [ ] Production: Migrate words
- [ ] Production: Migrate games & scores
- [ ] Production: Migrate app stats from realtime db
- [ ] Production: Migrate plans and plan_features (Remember to change the price_id to be the production one!)
- [ ] CICD - supabase deployment
